### ISSUE TITLE
Deployment Uploads fail on GlassFish with Grizzly error

### ISSUE THREAD

**Issue by glassfishrobot** [eclipse-ee4j/glassfish-grizzly#1713]
[https://java.net/jira/browse/GLASSFISH-21180](https://java.net/jira/browse/GLASSFISH-21180) is a related issue.

After extensive debugging by the Payara team our hunch is that the bug was introduced by commit

[https://github.com/GrizzlyNIO/grizzly-mirror/commit/262b9c90c5d9419163e078066ba7421a6d32b959](https://github.com/GrizzlyNIO/grizzly-mirror/commit/262b9c90c5d9419163e078066ba7421a6d32b959)

However this is not certain.

The symptoms are with secure admin enabled deploy to a GlassFish DAS
using

./asadmin deploy --force=true --upload=true /home/steve/Downloads/test-ejb.jar

the command fails with a timeout.

The timeout is caused because during mutipart mime parsing org.glassfish.grizzly.http.io.InputBuffer.fill is called which results in a new call to org.glassfish.grizzly.nio.tmpselectors.TemporarySelectorReader.read because the buffer does not have the required data and this read times out.

When the deployment is successful fill is not called and reads are not required.

As stated this is very time sensitve on the communication between the asadmin client and the server
#### Environment
GlassFish 4.1
#### Affected Versions
[2.3.15]

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
Reported by smillidge-c2b2

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
smillidge-c2b2 said:
Building a version of GlassFish against Grizzly 2.3.12 makes the deployment issue go away. However GlassFish then doesn't work for other SSL reasons.
I picked 2.3.12 as it was the version before the commit mentioned above.

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
smillidge-c2b2 said:
Deployment also works when built against 2.3.12, 2.3.13, 2.3.14.

2.3.15 fails with
HTTP connection failed with code 503, message: Service Unavailable
Command deploy failed.
2.3.15-gfa exhibits the problem.

I hope that narrows it down.

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
oleksiys said:
Can you pls. describe all the steps to reproduce the issue?

Thank you!

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
bornoz said:
Hi,
First, to be able to deploy application to a remote domain.(Only once)
sudo /opt/glassfish4/glassfish/bin/asadmin --host 192.168.11.10 --port 4848 login

Then deployment
sudo /opt/glassfish4/glassfish/bin/asadmin --host 192.168.11.10 --port 4848 redeploy --name myapplication /home/burak/myapplication.war

It works for GF4.0 without any problem.

Burak

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
smillidge-c2b2 said:
Hi bornoz

The problem is on GlassFish 4.1\. You must have secure admin enabled and you must use --upload=true in the deployment command.

See my next comment for how to reproduce

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
smillidge-c2b2 said:
Steps to reproduce

unzip glassfish-4.1.zip
cd glassfish4/bin/
./asadmin change-admin-password
./asadmin start-domain domain1
./asadmin login
./asadmin enable-secure-admin
./asadmin restart-domain domain1
./asadmin deploy --force=true --upload=true /home/steve/Downloads/test-ejb.jar
remote failure: Request failed.
Command deploy failed.

The first deploy may succeed but try a couple of times and it will fail consistently

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
smillidge-c2b2 said:
Server log exception of failed deployment

[2014-11-06T09:10:32.328+0000] [glassfish 4.1] [INFO] [NCLS-REST-00003] [javax.enterprise.admin.rest] [tid: _ThreadID=40 _ThreadName=admin-listener(2)] [timeMillis: 1415265032328] [levelValue: 800] [[

An error occurred while processing the request. Please see the server logs for details.

org.jvnet.mimepull.MIMEParsingException: java.io.IOException: java.util.concurrent.TimeoutException

at org.jvnet.mimepull.MIMEParser.fillBuf(MIMEParser.java:442)

at org.jvnet.mimepull.MIMEParser.skipPreamble(MIMEParser.java:307)

at org.jvnet.mimepull.MIMEParser.access$300(MIMEParser.java:68)

at org.jvnet.mimepull.MIMEParser$MIMEEventIterator.next(MIMEParser.java:149)

at org.jvnet.mimepull.MIMEParser$MIMEEventIterator.next(MIMEParser.java:132)

at org.jvnet.mimepull.MIMEMessage.makeProgress(MIMEMessage.java:198)

at org.jvnet.mimepull.MIMEMessage.parseAll(MIMEMessage.java:181)

at org.jvnet.mimepull.MIMEMessage.getAttachments(MIMEMessage.java:106)

at com.sun.enterprise.admin.remote.reader.MultipartProprietaryReader.readFrom(MultipartProprietaryReader.java:98)

at org.glassfish.admin.rest.readers.MultipartFDPayloadReader.readFrom(MultipartFDPayloadReader.java:72)

at org.glassfish.admin.rest.readers.MultipartFDPayloadReader.readFrom(MultipartFDPayloadReader.java:59)

at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.invokeReadFrom(ReaderInterceptorExecutor.java:251)

at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.aroundReadFrom(ReaderInterceptorExecutor.java:229)

at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor.proceed(ReaderInterceptorExecutor.java:149)

at org.glassfish.jersey.server.internal.MappableExceptionWrapperInterceptor.aroundReadFrom(MappableExceptionWrapperInterceptor.java:73)

at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor.proceed(ReaderInterceptorExecutor.java:149)

at org.glassfish.jersey.message.internal.MessageBodyFactory.readFrom(MessageBodyFactory.java:1124)

at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:851)

at org.glassfish.jersey.server.ContainerRequest.readEntity(ContainerRequest.java:270)

at org.glassfish.jersey.server.internal.inject.EntityParamValueFactoryProvider$EntityValueFactory.provide(EntityParamValueFactoryProvider.java:96)

at org.glassfish.jersey.server.spi.internal.ParameterValueHelper.getParameterValues(ParameterValueHelper.java:81)

at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$AbstractMethodParamInvoker.getParamValues(JavaResourceMethodDispatcherProvider.java:121)

at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:152)

at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:104)

at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:387)

at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:331)

at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:103)

at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:271)

at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)

at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)

at org.glassfish.jersey.internal.Errors.process(Errors.java:315)

at org.glassfish.jersey.internal.Errors.process(Errors.java:297)

at org.glassfish.jersey.internal.Errors.process(Errors.java:267)

at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:297)

at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:254)

at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1028)

at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:365)

at org.glassfish.admin.rest.adapter.JerseyContainerCommandService$3.service(JerseyContainerCommandService.java:173)

at org.glassfish.admin.rest.adapter.RestAdapter.service(RestAdapter.java:179)

at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:459)

at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:167)

at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:201)

at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:175)

at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:235)

at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)

at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)

at org.glassfish.grizzly.portunif.PUFilter.handleRead(PUFilter.java:231)

at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)

at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)

at org.glassfish.grizzly.portunif.PUFilter.handleRead(PUFilter.java:231)

at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)

at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)

at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:561)

at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:112)

at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:117)

at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:56)

at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:137)

at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:565)

at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:545)

at java.lang.Thread.run(Thread.java:744)

Caused by: java.io.IOException: java.util.concurrent.TimeoutException

at org.glassfish.grizzly.nio.transport.TCPNIOTransportFilter.handleRead(TCPNIOTransportFilter.java:90)

at org.glassfish.grizzly.filterchain.TransportFilter.handleRead(TransportFilter.java:173)

at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.read(DefaultFilterChain.java:351)

at org.glassfish.grizzly.filterchain.FilterChainContext.read(FilterChainContext.java:695)

at org.glassfish.grizzly.portunif.BackChannelFilter.handleRead(BackChannelFilter.java:80)

at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.read(DefaultFilterChain.java:351)

at org.glassfish.grizzly.filterchain.FilterChainContext.read(FilterChainContext.java:695)

at org.glassfish.grizzly.portunif.BackChannelFilter.handleRead(BackChannelFilter.java:80)

at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)

at org.glassfish.grizzly.filterchain.DefaultFilterChain.read(DefaultFilterChain.java:351)

at org.glassfish.grizzly.filterchain.FilterChainContext.read(FilterChainContext.java:695)

at org.glassfish.grizzly.http.io.InputBuffer.blockingRead(InputBuffer.java:1119)

at org.glassfish.grizzly.http.server.io.ServerInputBuffer.blockingRead(ServerInputBuffer.java:95)

at org.glassfish.grizzly.http.io.InputBuffer.fill(InputBuffer.java:1143)

at org.glassfish.grizzly.http.io.InputBuffer.read(InputBuffer.java:353)

at org.glassfish.grizzly.http.server.NIOInputStreamImpl.read(NIOInputStreamImpl.java:83)

at org.glassfish.jersey.message.internal.EntityInputStream.read(EntityInputStream.java:101)

at org.jvnet.mimepull.MIMEParser.fillBuf(MIMEParser.java:440)

... 71 more

Caused by: java.util.concurrent.TimeoutException

at org.glassfish.grizzly.nio.tmpselectors.TemporarySelectorReader.read(TemporarySelectorReader.java:126)

at org.glassfish.grizzly.nio.tmpselectors.TemporarySelectorReader.read(TemporarySelectorReader.java:75)

at org.glassfish.grizzly.AbstractReader.read(AbstractReader.java:72)

at org.glassfish.grizzly.nio.transport.TCPNIOTransportFilter.handleRead(TCPNIOTransportFilter.java:77)

... 96 more

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
smillidge-c2b2 said:
Grizzly Debug Trace of failed deploy after the SSL Handshake is complete

[2014-11-04T16:34:41.372+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.ssl.SSLConnectionContext] [tid: _ThreadID=99 _ThreadName=admin-listener(6)] [timeMillis: 1415118881372] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.ssl.SSLConnectionContext] [METHODNAME: wrap] [[
wrap done engine: 67476146[SSLEngine[hostname=null port=-1] TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA] result: Status = OK HandshakeStatus = FINISHED
bytesConsumed = 0 bytesProduced = 53 input: ByteBufferWrapper (644225315) [visible=[java.nio.HeapByteBuffer[pos=0 lim=0 cap=0]]] output: ByteBufferWrapper (2001033526) [visible=[java.nio.HeapByteBuffer[pos=0 lim=90 cap=33842]]]]]

[2014-11-04T16:34:41.373+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.nio.transport.TCPNIOTransport] [tid: _ThreadID=99 _ThreadName=admin-listener(6)] [timeMillis: 1415118881373] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.nio.transport.TCPNIOUtils] [METHODNAME: writeSimpleBuffer] [[
TCPNIOConnection (TCPNIOConnection{localSocketAddress=

{/127.0.0.1:4848}

, peerSocketAddress={/127.0.0.1:58340}}) (plain) write 90 bytes]]

[2014-11-04T16:34:41.373+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.ssl.SSLBaseFilter] [tid: _ThreadID=99 _ThreadName=admin-listener(6)] [timeMillis: 1415118881373] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.ssl.SSLBaseFilter] [METHODNAME: getPeerCertificates] [[
Error getting client certs
javax.net.ssl.SSLPeerUnverifiedException: peer not authenticated
at sun.security.ssl.SSLSessionImpl.getPeerCertificates(SSLSessionImpl.java:397)
at org.glassfish.grizzly.ssl.SSLBaseFilter.getPeerCertificates(SSLBaseFilter.java:880)
at org.glassfish.grizzly.ssl.SSLBaseFilter.getPeerCertificateChain(SSLBaseFilter.java:796)
at org.glassfish.grizzly.ssl.SSLBaseFilter.handleEvent(SSLBaseFilter.java:243)
at org.glassfish.grizzly.filterchain.ExecutorResolver$6.execute(ExecutorResolver.java:94)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
at org.glassfish.grizzly.filterchain.FilterChainContext.notifyDownstream(FilterChainContext.java:907)
at org.glassfish.grizzly.filterchain.FilterChainContext.notifyDownstream(FilterChainContext.java:890)
at org.glassfish.grizzly.portunif.BackChannelFilter.handleEvent(BackChannelFilter.java:131)
at org.glassfish.grizzly.filterchain.ExecutorResolver$6.execute(ExecutorResolver.java:94)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
at org.glassfish.grizzly.filterchain.FilterChainContext.notifyDownstream(FilterChainContext.java:907)
at org.glassfish.grizzly.filterchain.FilterChainContext.notifyDownstream(FilterChainContext.java:890)
at org.glassfish.grizzly.http.server.util.RequestUtils.populateCertificateAttribute(RequestUtils.java:75)
at org.glassfish.grizzly.http.server.Request.getAttribute(Request.java:873)
at org.glassfish.grizzly.http.server.Request.getUserPrincipal(Request.java:1859)
at com.sun.enterprise.admin.util.AdminCallbackHandler.<init>(AdminCallbackHandler.java:110)
at com.sun.enterprise.admin.util.GenericAdminAuthenticator.authenticate(GenericAdminAuthenticator.java:254)
at com.sun.enterprise.admin.util.GenericAdminAuthenticator.loginAsAdmin(GenericAdminAuthenticator.java:219)
at com.sun.enterprise.admin.util.GenericAdminAuthenticator.loginAsAdmin(GenericAdminAuthenticator.java:202)
at org.glassfish.admin.rest.adapter.RestAdapter.service(RestAdapter.java:168)
at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:459)
at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:167)
at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:201)
at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:175)
at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:235)
at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
at org.glassfish.grizzly.portunif.PUFilter.handleRead(PUFilter.java:231)
at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
at org.glassfish.grizzly.portunif.PUFilter.handleRead(PUFilter.java:231)
at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:561)
at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:112)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:117)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:56)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:137)
at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:565)
at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:545)
at java.lang.Thread.run(Thread.java:744)
]]

[2014-11-04T16:34:41.376+0000] [Payara 4.1] [FINER] [] [org.glassfish.grizzly.http.io.InputBuffer] [tid: _ThreadID=99 _ThreadName=admin-listener(6)] [timeMillis: 1415118881376] [levelValue: 400] [CLASSNAME: org.glassfish.grizzly.http.io.InputBuffer] [METHODNAME: log] [[
InputBuffer org.glassfish.grizzly.http.server.io.ServerInputBuffer@3bf214ba read byte array of len: 9219\. Ready content: BuffersBuffer (1434926288) [pos=0 lim=4614 cap=4614 bufferSize=4 buffers=[HeapBuffer (1649726074) [pos=4 lim=81 cap=81], HeapBuffer (564766606) [pos=4 lim=109 cap=109], HeapBuffer (154627302) [pos=5 lim=348 cap=348], HeapBuffer (365851844) [pos=5 lim=4094 cap=4094], null, null, null]]]]

[2014-11-04T16:34:41.377+0000] [Payara 4.1] [FINER] [] [org.glassfish.grizzly.http.io.InputBuffer] [tid: _ThreadID=99 _ThreadName=admin-listener(6)] [timeMillis: 1415118881377] [levelValue: 400] [CLASSNAME: org.glassfish.grizzly.http.io.InputBuffer] [METHODNAME: log] [[
InputBuffer org.glassfish.grizzly.http.server.io.ServerInputBuffer@3bf214ba read byte array of len: 4605\. Ready content: BuffersBuffer (1434926288) [pos=0 lim=0 cap=0 bufferSize=0 buffers=[null, null, null, null, null, null, null]]]]

[2014-11-04T16:34:41.377+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.websockets.BaseWebSocketFilter] [tid: _ThreadID=99 _ThreadName=admin-listener(6)] [timeMillis: 1415118881377] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.websockets.BaseWebSocketFilter] [METHODNAME: handleRead] [[
handleRead websocket: null content-size=4,089 headers=
HttpRequestPacket (
method=POST
url=/command/deploy
query=null
protocol=HTTP/1.1
content-length=-1
headers=[
authorization=Basic YWRtaW46cDBkdzBkMQ==
user-agent=hk2-agent
cache-control=no-cache
pragma=no-cache
x-if-command-model-match=vzKk4Gbhs7vnKex83EhEow==
x-requested-by=cli
accept=application/json; q=0.8, multipart/*; q=0.9
content-type=multipart/form-data; boundary=admfrmwrk_payload_2
mime-version=1.0
host=localhost:4848
connection=keep-alive
transfer-encoding=chunked]
)]]

[2014-11-04T16:34:41.378+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.websockets.BaseWebSocketFilter] [tid: _ThreadID=99 _ThreadName=admin-listener(6)] [timeMillis: 1415118881378] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.websockets.BaseWebSocketFilter] [METHODNAME: handleRead] [[
handleRead websocket: null content-size=4,089 headers=
HttpRequestPacket (
method=POST
url=/command/deploy
query=null
protocol=HTTP/1.1
content-length=-1
headers=[
authorization=Basic YWRtaW46cDBkdzBkMQ==
user-agent=hk2-agent
cache-control=no-cache
pragma=no-cache
x-if-command-model-match=vzKk4Gbhs7vnKex83EhEow==
x-requested-by=cli
accept=application/json; q=0.8, multipart/*; q=0.9
content-type=multipart/form-data; boundary=admfrmwrk_payload_2
mime-version=1.0
host=localhost:4848
connection=keep-alive
transfer-encoding=chunked]
)]]

[2014-11-04T16:34:41.379+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.nio.transport.TCPNIOTransport] [tid: _ThreadID=99 _ThreadName=admin-listener(6)] [timeMillis: 1415118881379] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.nio.transport.TCPNIOUtils] [METHODNAME: allocateAndReadBuffer] [[
TCPNIOConnection (TCPNIOConnection{localSocketAddress=

{/127.0.0.1:4848}

, peerSocketAddress={localhost/127.0.0.1:58340}}) (allocated) read 0 bytes]]

[2014-11-04T16:35:11.405+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.ssl.SSLConnectionContext] [tid: _ThreadID=99 _ThreadName=admin-listener(6)] [timeMillis: 1415118911405] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.ssl.SSLConnectionContext] [METHODNAME: wrap] [[
wrap engine: 67476146[SSLEngine[hostname=null port=-1] TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA] input: BuffersBuffer (376469260) [pos=1396 lim=1396 cap=1396 bufferSize=2 buffers=[HeapBuffer (832028758) [pos=0 lim=314 cap=314], HeapBuffer (1776559462) [pos=0 lim=1082 cap=1082], null, null]] output: ByteBufferWrapper (22249946) [visible=[java.nio.HeapByteBuffer[pos=1429 lim=33842 cap=33842]]]]]

[2014-11-04T16:35:11.406+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.ssl.SSLConnectionContext] [tid: _ThreadID=99 _ThreadName=admin-listener(6)] [timeMillis: 1415118911406] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.ssl.SSLConnectionContext] [METHODNAME: wrap] [[
wrap done engine: 67476146[SSLEngine[hostname=null port=-1] TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA] result: Status = OK HandshakeStatus = NOT_HANDSHAKING
bytesConsumed = 1396 bytesProduced = 1429 input: BuffersBuffer (376469260) [pos=0 lim=0 cap=0 bufferSize=0 buffers=[null, null, null, null]] output: ByteBufferWrapper (22249946) [visible=[java.nio.HeapByteBuffer[pos=0 lim=1429 cap=33842]]]]]

[2014-11-04T16:35:11.407+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.nio.transport.TCPNIOTransport] [tid: _ThreadID=99 _ThreadName=admin-listener(6)] [timeMillis: 1415118911407] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.nio.transport.TCPNIOUtils] [METHODNAME: writeSimpleBuffer] [[
TCPNIOConnection (TCPNIOConnection{localSocketAddress=

{/127.0.0.1:4848}

, peerSocketAddress={localhost/127.0.0.1:58340}}) (plain) write 1,429 bytes]]

[2014-11-04T16:35:11.408+0000] [Payara 4.1] [INFO] [NCLS-REST-00003] [javax.enterprise.admin.rest] [tid: _ThreadID=99 _ThreadName=admin-listener(6)] [timeMillis: 1415118911408] [levelValue: 800] [[
An error occurred while processing the request. Please see the server logs for details.
org.jvnet.mimepull.MIMEParsingException: java.io.IOException: java.util.concurrent.TimeoutException
at org.jvnet.mimepull.MIMEParser.fillBuf(MIMEParser.java:442)
at org.jvnet.mimepull.MIMEParser.skipPreamble(MIMEParser.java:307)
at org.jvnet.mimepull.MIMEParser.access$300(MIMEParser.java:68)
at org.jvnet.mimepull.MIMEParser$MIMEEventIterator.next(MIMEParser.java:149)
at org.jvnet.mimepull.MIMEParser$MIMEEventIterator.next(MIMEParser.java:132)
at org.jvnet.mimepull.MIMEMessage.makeProgress(MIMEMessage.java:198)
at org.jvnet.mimepull.MIMEMessage.parseAll(MIMEMessage.java:181)
at org.jvnet.mimepull.MIMEMessage.getAttachments(MIMEMessage.java:106)
at com.sun.enterprise.admin.remote.reader.MultipartProprietaryReader.readFrom(MultipartProprietaryReader.java:98)
at org.glassfish.admin.rest.readers.MultipartFDPayloadReader.readFrom(MultipartFDPayloadReader.java:72)
at org.glassfish.admin.rest.readers.MultipartFDPayloadReader.readFrom(MultipartFDPayloadReader.java:59)
at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.invokeReadFrom(ReaderInterceptorExecutor.java:251)
at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.aroundReadFrom(ReaderInterceptorExecutor.java:229)
at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor.proceed(ReaderInterceptorExecutor.java:149)
at org.glassfish.jersey.server.internal.MappableExceptionWrapperInterceptor.aroundReadFrom(MappableExceptionWrapperInterceptor.java:73)
at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor.proceed(ReaderInterceptorExecutor.java:149)
at org.glassfish.jersey.message.internal.MessageBodyFactory.readFrom(MessageBodyFactory.java:1124)
at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:851)
at org.glassfish.jersey.server.ContainerRequest.readEntity(ContainerRequest.java:270)
at org.glassfish.jersey.server.internal.inject.EntityParamValueFactoryProvider$EntityValueFactory.provide(EntityParamValueFactoryProvider.java:96)
at org.glassfish.jersey.server.spi.internal.ParameterValueHelper.getParameterValues(ParameterValueHelper.java:81)
at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$AbstractMethodParamInvoker.getParamValues(JavaResourceMethodDispatcherProvider.java:121)
at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:152)
at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:104)
at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:387)
at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:331)
at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:103)
at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:271)
at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:297)
at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:254)
at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1028)
at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:365)
at org.glassfish.admin.rest.adapter.JerseyContainerCommandService$3.service(JerseyContainerCommandService.java:173)
at org.glassfish.admin.rest.adapter.RestAdapter.service(RestAdapter.java:179)
at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:459)
at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:167)
at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:201)
at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:175)
at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:235)
at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
at org.glassfish.grizzly.portunif.PUFilter.handleRead(PUFilter.java:231)
at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
at org.glassfish.grizzly.portunif.PUFilter.handleRead(PUFilter.java:231)
at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:561)
at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:112)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:117)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:56)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:137)
at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:565)
at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:545)
at java.lang.Thread.run(Thread.java:744)
Caused by: java.io.IOException: java.util.concurrent.TimeoutException
at org.glassfish.grizzly.nio.transport.TCPNIOTransportFilter.handleRead(TCPNIOTransportFilter.java:90)
at org.glassfish.grizzly.filterchain.TransportFilter.handleRead(TransportFilter.java:173)
at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.read(DefaultFilterChain.java:351)
at org.glassfish.grizzly.filterchain.FilterChainContext.read(FilterChainContext.java:695)
at org.glassfish.grizzly.portunif.BackChannelFilter.handleRead(BackChannelFilter.java:80)
at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.read(DefaultFilterChain.java:351)
at org.glassfish.grizzly.filterchain.FilterChainContext.read(FilterChainContext.java:695)
at org.glassfish.grizzly.portunif.BackChannelFilter.handleRead(BackChannelFilter.java:80)
at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.read(DefaultFilterChain.java:351)
at org.glassfish.grizzly.filterchain.FilterChainContext.read(FilterChainContext.java:695)
at org.glassfish.grizzly.http.io.InputBuffer.blockingRead(InputBuffer.java:1119)
at org.glassfish.grizzly.http.server.io.ServerInputBuffer.blockingRead(ServerInputBuffer.java:95)
at org.glassfish.grizzly.http.io.InputBuffer.fill(InputBuffer.java:1143)
at org.glassfish.grizzly.http.io.InputBuffer.read(InputBuffer.java:353)
at org.glassfish.grizzly.http.server.NIOInputStreamImpl.read(NIOInputStreamImpl.java:83)
at org.glassfish.jersey.message.internal.EntityInputStream.read(EntityInputStream.java:101)
at org.jvnet.mimepull.MIMEParser.fillBuf(MIMEParser.java:440)
... 71 more
Caused by: java.util.concurrent.TimeoutException
at org.glassfish.grizzly.nio.tmpselectors.TemporarySelectorReader.read(TemporarySelectorReader.java:126)
at org.glassfish.grizzly.nio.tmpselectors.TemporarySelectorReader.read(TemporarySelectorReader.java:75)
at org.glassfish.grizzly.AbstractReader.read(AbstractReader.java:72)
at org.glassfish.grizzly.nio.transport.TCPNIOTransportFilter.handleRead(TCPNIOTransportFilter.java:77)
... 96 more
]]

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
smillidge-c2b2 said:
Grizzly Debug trace of a successful deploy after SSL Handshake

[2014-11-04T16:38:29.371+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.ssl.SSLConnectionContext] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109371] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.ssl.SSLConnectionContext] [METHODNAME: wrap] [[
wrap done engine: 61f37cfb[SSLEngine[hostname=null port=-1] TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA] result: Status = OK HandshakeStatus = FINISHED
bytesConsumed = 0 bytesProduced = 53 input: ByteBufferWrapper (644225315) [visible=[java.nio.HeapByteBuffer[pos=0 lim=0 cap=0]]] output: ByteBufferWrapper (372662894) [visible=[java.nio.HeapByteBuffer[pos=0 lim=90 cap=33842]]]]]

[2014-11-04T16:38:29.372+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.nio.transport.TCPNIOTransport] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109372] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.nio.transport.TCPNIOUtils] [METHODNAME: writeSimpleBuffer] [[
TCPNIOConnection (TCPNIOConnection{localSocketAddress=

{/127.0.0.1:4848}

, peerSocketAddress={/127.0.0.1:58356}}) (plain) write 90 bytes]]

[2014-11-04T16:38:29.372+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.ssl.SSLBaseFilter] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109372] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.ssl.SSLBaseFilter] [METHODNAME: getPeerCertificates] [[
Error getting client certs
javax.net.ssl.SSLPeerUnverifiedException: peer not authenticated
at sun.security.ssl.SSLSessionImpl.getPeerCertificates(SSLSessionImpl.java:397)
at org.glassfish.grizzly.ssl.SSLBaseFilter.getPeerCertificates(SSLBaseFilter.java:880)
at org.glassfish.grizzly.ssl.SSLBaseFilter.getPeerCertificateChain(SSLBaseFilter.java:796)
at org.glassfish.grizzly.ssl.SSLBaseFilter.handleEvent(SSLBaseFilter.java:243)
at org.glassfish.grizzly.filterchain.ExecutorResolver$6.execute(ExecutorResolver.java:94)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
at org.glassfish.grizzly.filterchain.FilterChainContext.notifyDownstream(FilterChainContext.java:907)
at org.glassfish.grizzly.filterchain.FilterChainContext.notifyDownstream(FilterChainContext.java:890)
at org.glassfish.grizzly.portunif.BackChannelFilter.handleEvent(BackChannelFilter.java:131)
at org.glassfish.grizzly.filterchain.ExecutorResolver$6.execute(ExecutorResolver.java:94)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
at org.glassfish.grizzly.filterchain.FilterChainContext.notifyDownstream(FilterChainContext.java:907)
at org.glassfish.grizzly.filterchain.FilterChainContext.notifyDownstream(FilterChainContext.java:890)
at org.glassfish.grizzly.http.server.util.RequestUtils.populateCertificateAttribute(RequestUtils.java:75)
at org.glassfish.grizzly.http.server.Request.getAttribute(Request.java:873)
at org.glassfish.grizzly.http.server.Request.getUserPrincipal(Request.java:1859)
at com.sun.enterprise.admin.util.AdminCallbackHandler.<init>(AdminCallbackHandler.java:110)
at com.sun.enterprise.admin.util.GenericAdminAuthenticator.authenticate(GenericAdminAuthenticator.java:254)
at com.sun.enterprise.admin.util.GenericAdminAuthenticator.loginAsAdmin(GenericAdminAuthenticator.java:219)
at com.sun.enterprise.admin.util.GenericAdminAuthenticator.loginAsAdmin(GenericAdminAuthenticator.java:202)
at org.glassfish.admin.rest.adapter.RestAdapter.service(RestAdapter.java:168)
at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:459)
at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:167)
at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:201)
at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:175)
at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:235)
at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
at org.glassfish.grizzly.portunif.PUFilter.handleRead(PUFilter.java:231)
at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
at org.glassfish.grizzly.portunif.PUFilter.handleRead(PUFilter.java:231)
at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:561)
at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:112)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:117)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:56)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:137)
at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:565)
at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:545)
at java.lang.Thread.run(Thread.java:744)
]]

[2014-11-04T16:38:29.375+0000] [Payara 4.1] [FINER] [] [org.glassfish.grizzly.http.io.InputBuffer] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109375] [levelValue: 400] [CLASSNAME: org.glassfish.grizzly.http.io.InputBuffer] [METHODNAME: log] [[
InputBuffer org.glassfish.grizzly.http.server.io.ServerInputBuffer@4da43159 read byte array of len: 9219\. Ready content: BuffersBuffer (165525050) [pos=0 lim=4614 cap=4614 bufferSize=4 buffers=[HeapBuffer (536916294) [pos=4 lim=81 cap=81], HeapBuffer (686407039) [pos=4 lim=109 cap=109], HeapBuffer (988192485) [pos=5 lim=348 cap=348], HeapBuffer (2060053144) [pos=5 lim=4094 cap=4094], null, null, null]]]]

[2014-11-04T16:38:29.375+0000] [Payara 4.1] [FINER] [] [org.glassfish.grizzly.http.io.InputBuffer] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109375] [levelValue: 400] [CLASSNAME: org.glassfish.grizzly.http.io.InputBuffer] [METHODNAME: log] [[
InputBuffer org.glassfish.grizzly.http.server.io.ServerInputBuffer@4da43159 read byte array of len: 4605\. Ready content: BuffersBuffer (165525050) [pos=0 lim=0 cap=0 bufferSize=0 buffers=[null, null, null, null, null, null, null]]]]

[2014-11-04T16:38:29.375+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.websockets.BaseWebSocketFilter] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109375] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.websockets.BaseWebSocketFilter] [METHODNAME: handleRead] [[
handleRead websocket: null content-size=4,089 headers=
HttpRequestPacket (
method=POST
url=/command/deploy
query=null
protocol=HTTP/1.1
content-length=-1
headers=[
authorization=Basic YWRtaW46cDBkdzBkMQ==
user-agent=hk2-agent
cache-control=no-cache
pragma=no-cache
x-if-command-model-match=vzKk4Gbhs7vnKex83EhEow==
x-requested-by=cli
accept=application/json; q=0.8, multipart/*; q=0.9
content-type=multipart/form-data; boundary=admfrmwrk_payload_4
mime-version=1.0
host=localhost:4848
connection=keep-alive
transfer-encoding=chunked]
)]]

[2014-11-04T16:38:29.376+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.websockets.BaseWebSocketFilter] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109376] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.websockets.BaseWebSocketFilter] [METHODNAME: handleRead] [[
handleRead websocket: null content-size=4,089 headers=
HttpRequestPacket (
method=POST
url=/command/deploy
query=null
protocol=HTTP/1.1
content-length=-1
headers=[
authorization=Basic YWRtaW46cDBkdzBkMQ==
user-agent=hk2-agent
cache-control=no-cache
pragma=no-cache
x-if-command-model-match=vzKk4Gbhs7vnKex83EhEow==
x-requested-by=cli
accept=application/json; q=0.8, multipart/*; q=0.9
content-type=multipart/form-data; boundary=admfrmwrk_payload_4
mime-version=1.0
host=localhost:4848
connection=keep-alive
transfer-encoding=chunked]
)]]

[2014-11-04T16:38:29.376+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.websockets.BaseWebSocketFilter] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109376] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.websockets.BaseWebSocketFilter] [METHODNAME: handleRead] [[
handleRead websocket: null content-size=4,089 headers=
HttpRequestPacket (
method=POST
url=/command/deploy
query=null
protocol=HTTP/1.1
content-length=-1
headers=[
authorization=Basic YWRtaW46cDBkdzBkMQ==
user-agent=hk2-agent
cache-control=no-cache
pragma=no-cache
x-if-command-model-match=vzKk4Gbhs7vnKex83EhEow==
x-requested-by=cli
accept=application/json; q=0.8, multipart/*; q=0.9
content-type=multipart/form-data; boundary=admfrmwrk_payload_4
mime-version=1.0
host=localhost:4848
connection=keep-alive
transfer-encoding=chunked]
)]]

[2014-11-04T16:38:29.376+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.websockets.BaseWebSocketFilter] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109376] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.websockets.BaseWebSocketFilter] [METHODNAME: handleRead] [[
handleRead websocket: null content-size=134 headers=
HttpRequestPacket (
method=POST
url=/command/deploy
query=null
protocol=HTTP/1.1
content-length=-1
headers=[
authorization=Basic YWRtaW46cDBkdzBkMQ==
user-agent=hk2-agent
cache-control=no-cache
pragma=no-cache
x-if-command-model-match=vzKk4Gbhs7vnKex83EhEow==
x-requested-by=cli
accept=application/json; q=0.8, multipart/*; q=0.9
content-type=multipart/form-data; boundary=admfrmwrk_payload_4
mime-version=1.0
host=localhost:4848
connection=keep-alive
transfer-encoding=chunked]
)]]

[2014-11-04T16:38:29.377+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.websockets.BaseWebSocketFilter] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109377] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.websockets.BaseWebSocketFilter] [METHODNAME: handleRead] [[
handleRead websocket: null content-size=27 headers=
HttpRequestPacket (
method=POST
url=/command/deploy
query=null
protocol=HTTP/1.1
content-length=-1
headers=[
authorization=Basic YWRtaW46cDBkdzBkMQ==
user-agent=hk2-agent
cache-control=no-cache
pragma=no-cache
x-if-command-model-match=vzKk4Gbhs7vnKex83EhEow==
x-requested-by=cli
accept=application/json; q=0.8, multipart/*; q=0.9
content-type=multipart/form-data; boundary=admfrmwrk_payload_4
mime-version=1.0
host=localhost:4848
connection=keep-alive
transfer-encoding=chunked]
)]]

[2014-11-04T16:38:29.377+0000] [Payara 4.1] [FINE] [] [org.glassfish.grizzly.websockets.BaseWebSocketFilter] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109377] [levelValue: 500] [CLASSNAME: org.glassfish.grizzly.websockets.BaseWebSocketFilter] [METHODNAME: handleRead] [[
handleRead websocket: null content-size=0 headers=
HttpRequestPacket (
method=POST
url=/command/deploy
query=null
protocol=HTTP/1.1
content-length=-1
headers=[
authorization=Basic YWRtaW46cDBkdzBkMQ==
user-agent=hk2-agent
cache-control=no-cache
pragma=no-cache
x-if-command-model-match=vzKk4Gbhs7vnKex83EhEow==
x-requested-by=cli
accept=application/json; q=0.8, multipart/*; q=0.9
content-type=multipart/form-data; boundary=admfrmwrk_payload_4
mime-version=1.0
host=localhost:4848
connection=keep-alive
transfer-encoding=chunked]
)]]

[2014-11-04T16:38:29.378+0000] [Payara 4.1] [FINER] [] [org.glassfish.grizzly.http.io.InputBuffer] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109378] [levelValue: 400] [CLASSNAME: org.glassfish.grizzly.http.io.InputBuffer] [METHODNAME: log] [[
InputBuffer org.glassfish.grizzly.http.server.io.ServerInputBuffer@4da43159 read byte array of len: 23\. Ready content: BuffersBuffer (1581834274) [pos=0 lim=7823 cap=7823 bufferSize=4 buffers=[HeapBuffer (988192485) [pos=521 lim=4094 cap=4094], HeapBuffer (1609139086) [pos=5 lim=4094 cap=4094], HeapBuffer (41093258) [pos=4 lim=138 cap=138], HeapBuffer (1312136519) [pos=4 lim=31 cap=31], null, null, null]]]]

[2014-11-04T16:38:29.378+0000] [Payara 4.1] [FINER] [] [org.glassfish.grizzly.http.io.InputBuffer] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109378] [levelValue: 400] [CLASSNAME: org.glassfish.grizzly.http.io.InputBuffer] [METHODNAME: log] [[
InputBuffer org.glassfish.grizzly.http.server.io.ServerInputBuffer@4da43159 read byte array of len: 48\. Ready content: BuffersBuffer (1581834274) [pos=0 lim=7800 cap=7800 bufferSize=4 buffers=[HeapBuffer (988192485) [pos=544 lim=4094 cap=4094], HeapBuffer (1609139086) [pos=5 lim=4094 cap=4094], HeapBuffer (41093258) [pos=4 lim=138 cap=138], HeapBuffer (1312136519) [pos=4 lim=31 cap=31], null, null, null]]]]

[2014-11-04T16:38:29.378+0000] [Payara 4.1] [FINER] [] [org.glassfish.grizzly.http.io.InputBuffer] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109378] [levelValue: 400] [CLASSNAME: org.glassfish.grizzly.http.io.InputBuffer] [METHODNAME: log] [[
InputBuffer org.glassfish.grizzly.http.server.io.ServerInputBuffer@4da43159 read byte array of len: 29\. Ready content: BuffersBuffer (1581834274) [pos=0 lim=7752 cap=7752 bufferSize=4 buffers=[HeapBuffer (988192485) [pos=592 lim=4094 cap=4094], HeapBuffer (1609139086) [pos=5 lim=4094 cap=4094], HeapBuffer (41093258) [pos=4 lim=138 cap=138], HeapBuffer (1312136519) [pos=4 lim=31 cap=31], null, null, null]]]]

[2014-11-04T16:38:29.379+0000] [Payara 4.1] [FINER] [] [org.glassfish.grizzly.http.io.InputBuffer] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109379] [levelValue: 400] [CLASSNAME: org.glassfish.grizzly.http.io.InputBuffer] [METHODNAME: log] [[
InputBuffer org.glassfish.grizzly.http.server.io.ServerInputBuffer@4da43159 read byte array of len: 50\. Ready content: BuffersBuffer (1581834274) [pos=0 lim=7723 cap=7723 bufferSize=4 buffers=[HeapBuffer (988192485) [pos=621 lim=4094 cap=4094], HeapBuffer (1609139086) [pos=5 lim=4094 cap=4094], HeapBuffer (41093258) [pos=4 lim=138 cap=138], HeapBuffer (1312136519) [pos=4 lim=31 cap=31], null, null, null]]]]

[2014-11-04T16:38:29.379+0000] [Payara 4.1] [FINER] [] [org.glassfish.grizzly.http.io.InputBuffer] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109379] [levelValue: 400] [CLASSNAME: org.glassfish.grizzly.http.io.InputBuffer] [METHODNAME: log] [[
InputBuffer org.glassfish.grizzly.http.server.io.ServerInputBuffer@4da43159 read byte array of len: 55\. Ready content: BuffersBuffer (1581834274) [pos=0 lim=7673 cap=7673 bufferSize=4 buffers=[HeapBuffer (988192485) [pos=671 lim=4094 cap=4094], HeapBuffer (1609139086) [pos=5 lim=4094 cap=4094], HeapBuffer (41093258) [pos=4 lim=138 cap=138], HeapBuffer (1312136519) [pos=4 lim=31 cap=31], null, null, null]]]]

[2014-11-04T16:38:29.379+0000] [Payara 4.1] [FINER] [] [org.glassfish.grizzly.http.io.InputBuffer] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109379] [levelValue: 400] [CLASSNAME: org.glassfish.grizzly.http.io.InputBuffer] [METHODNAME: log] [[
InputBuffer org.glassfish.grizzly.http.server.io.ServerInputBuffer@4da43159 read byte array of len: 320\. Ready content: BuffersBuffer (1581834274) [pos=0 lim=7618 cap=7618 bufferSize=4 buffers=[HeapBuffer (988192485) [pos=726 lim=4094 cap=4094], HeapBuffer (1609139086) [pos=5 lim=4094 cap=4094], HeapBuffer (41093258) [pos=4 lim=138 cap=138], HeapBuffer (1312136519) [pos=4 lim=31 cap=31], null, null, null]]]]

[2014-11-04T16:38:29.380+0000] [Payara 4.1] [FINER] [] [org.glassfish.grizzly.http.io.InputBuffer] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109380] [levelValue: 400] [CLASSNAME: org.glassfish.grizzly.http.io.InputBuffer] [METHODNAME: log] [[
InputBuffer org.glassfish.grizzly.http.server.io.ServerInputBuffer@4da43159 read byte array of len: 8192\. Ready content: BuffersBuffer (1581834274) [pos=0 lim=7298 cap=7298 bufferSize=4 buffers=[HeapBuffer (988192485) [pos=1046 lim=4094 cap=4094], HeapBuffer (1609139086) [pos=5 lim=4094 cap=4094], HeapBuffer (41093258) [pos=4 lim=138 cap=138], HeapBuffer (1312136519) [pos=4 lim=31 cap=31], null, null, null]]]]

[2014-11-04T16:38:29.380+0000] [Payara 4.1] [FINER] [] [org.glassfish.grizzly.http.io.InputBuffer] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109380] [levelValue: 400] [CLASSNAME: org.glassfish.grizzly.http.io.InputBuffer] [METHODNAME: log] [[
InputBuffer org.glassfish.grizzly.http.server.io.ServerInputBuffer@4da43159 read byte array of len: 894\. Ready content: BuffersBuffer (1581834274) [pos=0 lim=0 cap=0 bufferSize=0 buffers=[null, null, null, null, null, null, null]]]]

[2014-11-04T16:38:29.495+0000] [Payara 4.1] [INFO] [] [javax.enterprise.system.tools.deployment.common] [tid: _ThreadID=176 _ThreadName=admin-listener(10)] [timeMillis: 1415119109495] [levelValue: 800] [[
visiting unvisited references]]

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
smillidge-c2b2 said:
Placing a breakpoint in org.glassfish.grizzly.websockets.BaseWebSocketFilter.handleRead or anywhere in the read processing will hit during authentication. After the breakpoint has been hit the deployment will always be successful.

Placing the breakpoint in org.jvnet.mimepull.MIMEParser.fillBuf
The deployment will fail when the parser tries to read more data off the socket.

I suspect there is a horrible race condition somewhere as many times deploying from within asadmin i.e.

./asadmin
asadmin>
asadmin> deploy --force=true --upload=true /home/steve/Downloads/echo.simple-0.0.1-SNAPSHOT.jar

will work sometimes and not others.

Also disabling secure-admin will also cause deployments to start working. However remote deployments can not be carried out with secure-admin disabled

With secure-admin enabled the asadmin CLI first tries to post the multip-part upload via HTTP it hten receives a redirect to HTTPS and repeats the POST over HTTPS. This is the main difference between secure and unsecure. This may be significant or it could be that the SSL negotiation just adds in a tining difference.

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
bornoz said:
I am afraid we have to wait for a bug-fix. Right?

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
oleksiys said:
pls. try this patch and let me know if it works for you.

thank you.
[https://dl.dropboxusercontent.com/u/7319744/glassfish-4.1/nucleus-grizzly-all.jar](https://dl.dropboxusercontent.com/u/7319744/glassfish-4.1/nucleus-grizzly-all.jar)

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
smillidge-c2b2 said:
Thanks Fantastic response.

The patch works for me both locally with --upload=true and from a host remote from the DAS.

Any chance you can commit the fix onto the GitHub branch or even better release a patched 2_3_15_gfa to maven central?

Thanks

Steve

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
oleksiys said:
the issue has been already fixed on 2.3.16.
fixed on glassfishv41 branch.

Revision: 62036e9dfa64d0db70fe9b39059ea18aebed80c8
Date: 2014-11-10 23:30:00 UTC

Log Message:
------------
[glassfishv4] + fix issue #1713
[https://java.net/jira/browse/GRIZZLY-1713](https://java.net/jira/browse/GRIZZLY-1713)

"Deployment Uploads fail on GlassFish with Grizzly error"

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
Marked as **fixed** on Monday, November 10th 2014, 3:52:57 pm

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
bornoz said:
Thanks oleksiys,
It worked for me as well.

Cheers

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
lukaskuzmiak said:
fixes remote deployment by NetBeans - everything suddenly works flawlessly - patching server only seems to be enough (even though there are glassfish libraries on client for netbeans too).

Thanks a lot!
Cheers,
Lukas

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
gcruscoe said:
oleksiys, does this patched nucleus-grizzly-all.jar include the fix for bug 21118 ([https://java.net/jira/browse/GLASSFISH-21118)?](https://java.net/jira/browse/GLASSFISH-21118)?) There was a need for a patched jar of nucleus-grizzly-all.jar for that bug as well. Trying to verify whether this version would contain both fixes?

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
oleksiys said:
yes, it's the complete patch for known 4.1 bugs (grizzly related)

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
tovine said:
The patched nucleus-grizzly-all.jar doesn't play nice with the recent nightly builds of glassfish (late february -15 and newer), how can I compile a newer version myself? I've already cloned the grizzly-mirror repo from github.

Sorry if this is a 'noob question' with an obvious answer - I feel like I've been searching all over the web several times without finding it...

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
smillidge-c2b2 said:
Shameless plug but try this [http://www.payara.co/downloads](http://www.payara.co/downloads) as it incorporates the patches.

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
oleksiys said:
I'm planning to make another integration soon, pls. stay tuned.

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
tovine said:
smillidge-c2b2: thanks a lot, Payara worked really well (except for remote deployments, a new issue now: 'I/O Error: Error writing request body to server' after it's done uploading the file) so we'll just go with that for now. By the way - here's a workaround that works (it's in two steps and not ideal, but better than the alternative):

scp [app_archive] user@remote:/tmp/ && ssh user@remote /path/to/asadmin -u admin --passwordfile /path/to/passwdfile deploy --force=true /tmp/[app_archive]

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
@arindam-bandyopadhyay said:
Grizzly 2.3.19 integrated with GF trunk/main. on 8th April 2015.
Committed revision 63835
It fixes the issue in recent nightly builds of glassfish .

**Comment by glassfishrobot** [comment]
@glassfishrobot Commented 
This issue was imported from java.net JIRA GRIZZLY-1713

**Comment by glassfishrobot** [comment]
* **Issue Imported From:** https://github.com/javaee/grizzly/issues/1713
* **Original Issue Raised By:**@glassfishrobot
* **Original Issue  Assigned To:** @glassfishrobot
* **Original Issue  Closed By:**@glassfishrobot
